pkg/cli/cmd/resource/create/create.go

```diff
@@ -0,0 +1,155 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/radius-project/radius/pkg/cli"
+	"github.com/radius-project/radius/pkg/cli/clients_new/generated"
+	"github.com/radius-project/radius/pkg/cli/clierrors"
+	"github.com/radius-project/radius/pkg/cli/cmd/commonflags"
+	"github.com/radius-project/radius/pkg/cli/cmd/resourceprovider/common"
+	"github.com/radius-project/radius/pkg/cli/connections"
+	"github.com/radius-project/radius/pkg/cli/framework"
+	"github.com/radius-project/radius/pkg/cli/output"
+	"github.com/radius-project/radius/pkg/cli/workspaces"
+	"github.com/spf13/cobra"
+)
+
+// NewCommand creates an instance of the `rad resourceprovider create` command and runner.
+func NewCommand(factory framework.Factory) (*cobra.Command, framework.Runner) {
+	runner := NewRunner(factory)
+
+	cmd := &cobra.Command{
+		Use:   "create [resource type] [name] [input]",
+		Short: "Create or update a resource",
+		Long: `Create or update a resource
+		
+Resources are the entities are the primary entities that make up applactions.
+
+Input can be passed in using a file or inline JSON as the second argument. Prefix the input with '@' to indicate a file path.
+`,
+		Example: `
+# Create a resource (from file)
+rad resource create 'Applications.Core/containers' mycontainer @/path/to/input.json
+
+# Create a resource (inline)
+rad resourceprovider create 'Applications.Core/containers' mycontainer '{ ... }'`,
+		Args: cobra.ExactArgs(3),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddOutputFlag(cmd)
+	commonflags.AddWorkspaceFlag(cmd)
+
+	return cmd, runner
+}
+
+// Runner is the Runner implementation for the `rad resourceprovider create` command.
+type Runner struct {
+	ConnectionFactory connections.Factory
+	ConfigHolder      *framework.ConfigHolder
+	Output            output.Interface
+	Format            string
+	Workspace         *workspaces.Workspace
+
+	ResourceType string
+	ResourceName string
+	Resource     *generated.GenericResource
+}
+
+// NewRunner creates an instance of the runner for the `rad resourceprovider create` command.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConnectionFactory: factory.GetConnectionFactory(),
+		ConfigHolder:      factory.GetConfigHolder(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad resourceprovider create` command.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	// Validate command line args and
+	workspace, err := cli.RequireWorkspace(cmd, r.ConfigHolder.Config, r.ConfigHolder.DirectoryConfig)
+	if err != nil {
+		return err
+	}
+	r.Workspace = workspace
+
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	r.ResourceType = args[0]
+	r.ResourceName = args[1]
+	r.Resource, err = readInput(args[2])
```

Comment:
Should we check if there are 3 arguments first of all? Should we check if the ResourceType exists for example?

Reply:
No we don't need to. Cobra does it for us.